base/essentials.jl

```diff
@@ -148,6 +148,24 @@ macro boundscheck(blk)
       $(Expr(:boundscheck,:pop)))
 end
 
+"""
+    @inbounds(blk)
+
+In the example below the bound check of array A is skipped to improve performance.
+```julia
+    function sum(A::AbstractArray)
+    r = zero(eltype(A))
+    for i = 1:length(A)
+    @inbounds r += A[i]
+    end
+    return r
+    end
```

Comment:
I probably didn't explain in https://github.com/JuliaLang/julia/pull/19726#discussion_r94218916 very clearly, sorry about that @Shade5. When I wrote "won't need to be indented" what I meant was to just remove the left-most 4 spaces of indentation, not all of the indentation. So this should just be written as

    ```julia
    function sum(A::AbstractArray)
        r = zero(eltype(A))
        for i = 1:length(A)
            @inbounds r += A[i]
        end
        return r
    end
    ```

Same with the other code block below.